src/lotus_json/mod.rs

```diff
@@ -347,34 +347,39 @@ pub mod hexify_bytes {
 
 pub mod hexify_vec_bytes {
     use super::*;
-    use std::fmt::Write;
+    use std::borrow::Cow;
 
     pub fn serialize<S>(value: &[u8], serializer: S) -> Result<S::Ok, S::Error>
     where
         S: Serializer,
     {
-        let mut s = String::with_capacity(2 + value.len() * 2);
-        s.push_str("0x");
-        for b in value {
-            write!(s, "{:02x}", b).expect("failed to write to string");
-        }
-        serializer.serialize_str(&s)
+        let mut s = vec![0; 2 + value.len() * 2];
+        s.get_mut(0..2)
+            .expect("len is correct")
+            .copy_from_slice(b"0x");
+        hex::encode_to_slice(value, s.get_mut(2..).expect("len is correct"))
+            .map_err(serde::ser::Error::custom)?;
+        serializer.serialize_str(std::str::from_utf8(&s).expect("valid utf8"))
     }
 
     pub fn deserialize<'de, D>(deserializer: D) -> Result<Vec<u8>, D::Error>
     where
         D: Deserializer<'de>,
     {
         let s = String::deserialize(deserializer)?;
-        if (s.len() >= 2 && s.len() % 2 == 0) && s.get(..2).expect("failed to get prefix") == "0x" {
-            let result: Result<Vec<u8>, _> = (2..s.len())
-                .step_by(2)
-                .map(|i| u8::from_str_radix(s.get(i..i + 2).expect("failed to get slice"), 16))
-                .collect();
-            result.map_err(serde::de::Error::custom)
+        let s = Cow::from(s.strip_prefix("0x").unwrap_or(&s));
+
+        // Pad with 0 if odd length. This is necessary because [`hex::decode`] requires an even
+        // number of characters, whereas a valid input is also `0x0`.
+        let s = if s.len() % 2 == 0 {
+            s
         } else {
-            Err(serde::de::Error::custom("Invalid hex"))
-        }
+            let mut s = s.into_owned();
+            s.insert(0, '0');
```

Comment:
This may allocate memory and involve some copying, but I think it's acceptable in this context

Reply:
That's the point of using the Cow, to limit the allocation in case it's not needed. :)